torch/nn/quantized/modules/linear.py

```diff
@@ -36,7 +36,7 @@ class Linear(torch.nn.Module):
     """
     _FLOAT_MODULE = nn.Linear
 
-    def __init__(self, in_features, out_features, bias_=True):
+    def __init__(self, in_features, out_features, bias_=True, dtype=torch.qint8):
```

Comment:
this seems to be changing the static quantization module, do we have static quantization kernel support for fp16 data?
Also can you add tests for this change and change the commit message to reflect this?

Reply:
So long as the dynamic quantization module inherits from this module, this module needs to have the APIs necessary to support dynamic quantization, or we can make the dynamic module not inherit from this one.

As it is, `forward()` of the static module should fail if fp16 packed weights are given to it

Reply:
can we split the common part to another module then?